master/internal/db/postgres_tasks.go

```diff
@@ -437,6 +438,28 @@ WHERE task_id = $1
 	return count, nil
 }
 
+// DeleteExpiredTaskLogs deletes task logs older than the log_retention_days time where defined and
+// non-negative. If log_retention is not defined the defaultLogRetention value is used instead.
+func (db *PgDB) DeleteExpiredTaskLogs(defaultLogRetentionDays int16) (int64, error) {
+	r, err := db.sql.Exec(fmt.Sprintf(`
+WITH log_retention_tasks AS (
+	SELECT task_id, end_time, COALESCE(log_retention_days, %d) AS current_retention_days FROM tasks
+	WHERE task_id IN (SELECT DISTINCT task_id FROM task_logs)
```

Comment:
correct, it seems like this would help if we have a database with a very large number of tasks but fewer task logs, which is likely to be the situation once retention is enabled.

this might be a good candidate for scale testing & optimization tho.